browser/brave_browser_main_parts.cc

```diff
@@ -197,7 +226,8 @@ void BraveBrowserMainParts::PostProfileInit(Profile* profile,
     extensions::ExtensionService* extension_service =
         extensions::ExtensionSystem::Get(profile)->extension_service();
     DCHECK(extension_service);
-    greaselion_service->SetExtensionService(extension_service);
+    greaselion_service->SetDelegate(
+        std::make_unique<GreaselionServiceDelegateImpl>(extension_service));
```

Comment:
a better approach here would be to create this delegate in `GreaselionServiceFactory`, pass into it `profile/browser_context` and request `extension_service` only at `AddExtension` or `UnloadExtension` calls (if `extension_service_` is `nullptr`).

Reply:
Good call